ykrt/src/compile/jitc_yk/opt/mod.rs

```diff
@@ -2134,6 +2136,81 @@ mod test {
         );
     }
 
+    #[test]
+    fn opt_peeling_heap() {
+        // Loads
+        Module::assert_ir_transform_eq(
+            "
+          entry:
+            %0: ptr = param reg
+            header_start [%0]
+            %2: i8 = load %0
+            %3: i1 = eq %2, 1i8
+            guard true, %3, []
+            %5: ptr = ptr_add %0, 1
+            %6: i8 = load %5
+            black_box %2
+            black_box %6
+            header_end [%0]
+        ",
+            |m| opt(m).unwrap(),
+            "
+          ...
+            body_start [%10]
+            %15: ptr = ptr_add %10, 1
+            %16: i8 = load %15
+            black_box 1i8
+            black_box %16
```

Comment:
So is this test showing that it's not necessary to include `%2: i8 = load %0` in the body of the loop because `%2` is known to be `1`?

Reply:
Exactly.